### PR TITLE
test: remove jest `globals` config

### PR DESCRIPTION
**Pull request by yin1999**
## Summary

Fixes: #7560

### Solution

use transorm instead, see: <https://jestjs.io/docs/code-transformation#defaults>

---

## Screenshots

None

---

## How did you test this change?

run `yarn test`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -49,16 +49,19 @@
...
-    ]
+    ],
+    "transform": {
+      "\\.[jt]sx?$": [
+        "ts-jest",
+        {
+          "babelConfig": true
+        }
+      ]
+    }
```
Comment (yin1999) [inline package.json:64]: https://kulshekhar.github.io/ts-jest/docs/getting-started/options/babelConfig/

File: tsconfig.json [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {
   "compilerOptions": {
+    "allowJs": true,
```
Comment (yin1999) [inline tsconfig.json:3]: avoid warning

image

**Review by caugner — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -49,16 +49,19 @@
...
-      }
-    },
     "preset": "ts-jest",
     "testPathIgnorePatterns": [
       "headless*",
       "developing.spec.ts"
-    ]
+    ],
+    "transform": {
+      "\\.[jt]sx?$": [
```
[the remark this answers is not in the document]
Comment (yin1999) [reply package.json]: It works

**Comment by caugner** [comment]
Thank you @yin1999! 🎉

**Pull request merged**